sklearn/metrics/tests/test_regression.py

```diff
@@ -34,6 +36,25 @@ def test_regression_metrics(n_samples=50):
     assert_almost_equal(max_error(y_true, y_pred), 1.)
     assert_almost_equal(r2_score(y_true, y_pred),  0.995, 2)
     assert_almost_equal(explained_variance_score(y_true, y_pred), 1.)
+    assert_almost_equal(mean_tweedie_deviance_error(y_true, y_pred, p=0),
+                        mean_squared_error(y_true, y_pred))
+
+    # Tweedie deviance needs positive y_pred, except for p=0,
+    # p>=2 needs positive y_true
+    # results evaluated by sympy
+    y_true = np.arange(1, 1 + n_samples)
+    y_pred = 2 * y_true
+    n = n_samples
+    assert_almost_equal(mean_tweedie_deviance_error(y_true, y_pred, p=-1),
+                        5/12 * n * (n**2 + 2 * n + 1))
+    assert_almost_equal(mean_tweedie_deviance_error(y_true, y_pred, p=1),
+                        (n + 1) * (1 - np.log(2)))
+    assert_almost_equal(mean_tweedie_deviance_error(y_true, y_pred, p=2),
+                        2 * np.log(2) - 1)
+    assert_almost_equal(mean_tweedie_deviance_error(y_true, y_pred, p=3/2),
+                        (6 * np.sqrt(2) - 8)/n * np.sqrt(y_true).sum())
+    assert_almost_equal(mean_tweedie_deviance_error(y_true, y_pred, p=3),
+                        np.sum(1 / y_true) / (4 * n))
```

Comment:
Could you please add a new tests to check that the continuity at the 4 boundaries: p=0-, p=1+ and p=2- and p=2+:

```python
 assert_almost_equal(mean_tweedie_deviance_error(y_true, y_pred, p=0 - 1e-12),
                     mean_tweedie_deviance_error(y_true, y_pred, p=0))

 assert_almost_equal(mean_tweedie_deviance_error(y_true, y_pred, p=1 + 1e-12),
                     mean_tweedie_deviance_error(y_true, y_pred, p=1))

 assert_almost_equal(mean_tweedie_deviance_error(y_true, y_pred, p=2 - 1e-12),
                     mean_tweedie_deviance_error(y_true, y_pred, p=2))

 assert_almost_equal(mean_tweedie_deviance_error(y_true, y_pred, p=2 + 1e-12),
                     mean_tweedie_deviance_error(y_true, y_pred, p=2))
```


One might need to set some suitable tolerance levels for the checks.

Reply:
Good idea indeed!